evm/src/cpu/kernel/asm/rlp/read_to_memory.asm

```diff
@@ -13,29 +13,26 @@ global read_rlp_to_memory:
 
     PUSH @SEGMENT_RLP_RAW // ctx == virt == 0
     // stack: addr, final_addr, retdest
-
 read_rlp_to_memory_loop:
     // stack: addr, final_addr, retdest
     DUP2
     DUP2
-    EQ
-    // stack: addr == final_addr, addr, final_addr, retdest
+    LT
+    ISZERO
+    // stack: addr >= final_addr, addr, final_addr, retdest
     %jumpi(read_rlp_to_memory_finish)
-    // stack: addr, len, retdest
-    DUP1
-    PROVER_INPUT(rlp)
-    // stack: byte, addr, addr, final_addr, retdest
-    MSTORE_GENERAL
     // stack: addr, final_addr, retdest
-    %increment
+    PROVER_INPUT(rlp)
+    SWAP1
+    MSTORE_32BYTES_32
     // stack: addr', final_addr, retdest
     %jump(read_rlp_to_memory_loop)
 
 read_rlp_to_memory_finish:
     // stack: addr, final_addr, retdest
     // we recover the offset here
     PUSH @SEGMENT_RLP_RAW // ctx == virt == 0
-    DUP2 SUB
+    DUP3 SUB
```

Comment:
this is changed but no stack doc is changed, so `pos` now is referring to the final offset instead of the current one?

Reply:
In the original version the condition for exiting the loop was `addr == final_addr`. Since now is possible that `addr 
 > final_add`, which happens when the last chunk received from the prover inputs  is padded with 0s, the final positon to be returned is `final_addr`